teuthology/orchestra/run.py

```diff
@@ -268,15 +272,39 @@ def copy_to_log(f, logger, loglevel=logging.INFO, capture=None):
     # Work-around for http://tracker.ceph.com/issues/8313
     if isinstance(f, ChannelFile):
         f._flags += ChannelFile.FLAG_BINARY
-
     for line in f:
         if capture:
-            capture.write(line)
+            #out = stringify(line)
+            if PY3:
+                if isinstance(line, str):
+                    out = line.encode()
+                else:
+                    out = line.decode()
+                if isinstance(capture, io.StringIO):
+                    capture.write(out)
+                elif isinstance(capture, io.BytesIO):
+                    capture.write(out.encode())
+            else:
+                if isinstance(line, str):
+                    out = line.decode()
+                else:
+                    out = line.encode()
+                if isinstance(capture, io.StringIO):
+                    capture.write(out)
+                elif isinstance(capture, io.BytesIO):
+                    capture.write(out.encode())
+                else:
+                    # isinstance does not work with cStringIO.StringIO and
+                    # fails with error:
+                    #   TypeError: isinstance() arg 2 must be a class, type,
+                    #   or tuple of classes and types
+                    capture.write(out.encode())
         line = line.rstrip()
         # Second part of work-around for http://tracker.ceph.com/issues/8313
         try:
-            line = unicode(line, 'utf-8', 'replace').encode('utf-8')
-            logger.log(loglevel, line.decode('utf-8'))
+            if isinstance(line, bytes):
```

Comment:
could use `six.ensure_str()`.

Reply:
At the moment of writing code there were no six.ensure_str method.